h2o-algos/src/main/java/hex/tree/isoforextended/ExtendedIsolationForest.java

```diff
@@ -0,0 +1,324 @@
+package hex.tree.isoforextended;
+
+import hex.ModelCategory;
+import hex.ScoreKeeper;
+import hex.psvm.psvm.MatrixUtils;
+import hex.tree.SharedTree;
+import water.*;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.fvec.Vec;
+import water.util.ArrayUtils;
+import water.util.MathUtils;
+import water.util.VecUtils;
+
+/**
+ * Extended isolation forest implementation. Algorithm comes from https://arxiv.org/pdf/1811.02141.pdf paper.
+ *
+ * @author Adam Valenta
+ */
+public class ExtendedIsolationForest extends SharedTree<ExtendedIsolationForestModel,
```

Comment:
what capabilities do we get from SharedTree?

Reply:
this question is even more relevant now since scoreAndBuildTrees is currently not used

Reply:
The major reason is to be able to follow Isolation Forest as a referential implementation and lack of knowledge of the codebase. Therefore I used the same parent class. Currently, I am not using any of the capabilities besides the interface that allows me to save, load model, and generate interfaces the same way as it is for Isolation Forest. Do you think it would be doable and better to extend Model directly?

Reply:
The major reason is to be able to follow Isolation Forest as a referential implementation and lack of knowledge of the codebase. Therefore I used the same parent class. Currently, I am not using any of the capabilities besides the interface that allows me to save, load model, and generate interfaces the same way as it is for Isolation Forest. Do you think it would be doable and better to extend Model directly?

Reply:
Resolved in ff3dcb86a04ea08f4b6373fc1dcf2abcd751f13e on PR: https://github.com/h2oai/h2o-3/pull/4907